### PR TITLE
Implement ThreadRepliesCounter decoration

### PR DESCRIPTION
**Pull request by ipavlidakis**
### 🔗 Issue Links

_Provide all Jira tickets and/or Github issues related to this PR, if applicable._

### 🎯 Goal

Add a a counter of replies in a thread that is being displayed below the first(source) message.

### 📝 Summary

Leverage the new decorations views API to implement the counter. As we would like the counter to appear below the first(source) message, we are setting it up as a footer.

### 🛠 Implementation

_Provide a detailed description of the implementation and explain your decisions if you find them relevant._

### 🎨 Showcase

|Without Replies|With Replies|
|---------------|------------|
|Untitled|thread-list-replies-counter-decoration-enabled|

### 🧪 Manual Testing Notes

_Explain how this change can be tested manually, if applicable._

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change follows zero ⚠️ policy (required)
- [x] This change should be manually QAed
- [x] Changelog is updated with client-facing changes
- [x] New code is covered by unit tests
- [x] Comparison screenshots added for visual changes
- [x] Affected documentation updated (docusaurus, tutorial, CMS)

### 🎁 Meme

_Provide a funny gif or image that relates to your work on this pull request. (Optional)_

### CODE REVIEW

File: Sources/StreamChatUI/ChatThread/ChatThreadVC.swift [hunk 1]
```diff
@@ -258,6 +258,26 @@ open class ChatThreadVC: _ViewController,
...
+        decorationViewForMessage message: ChatMessage,
+        decorationType: ChatMessageDecorationType,
+        at indexPath: IndexPath
+    ) -> ChatMessageDecorationView? {
+        switch decorationType {
+        case .header where vc.shouldShowDateSeparator(forMessage: message, at: indexPath):
+            let dateSeparatorView = components.messageListDateSeparatorView.init()
+            dateSeparatorView.content = vc.dateSeparatorFormatter.format(message.createdAt)
+            return dateSeparatorView
+        case .footer where components.threadRepliesCounterEnabled && message == messages.last:
```
Comment (polqf) [inline Sources/StreamChatUI/ChatThread/ChatThreadVC.swift:272]: Yeah, I think this verifies what I said in the previous PR. IMO having footer and header in separate delegate methods makes more sense

File: Sources/StreamChatUI/Utils/Extensions/UIView+Extensions.swift [hunk 2]
```diff
@@ -7,8 +7,7 @@ import UIKit
 extension UIView {
     // MARK: - `embed` family of helpers
 
-    @objc
-    func embed(_ subview: UIView, insets: NSDirectionalEdgeInsets = .zero) {
+    @objc func embed(_ subview: UIView, insets: NSDirectionalEdgeInsets = .zero) {
```
Comment (ipavlidakis) [inline Sources/StreamChatUI/Utils/Extensions/UIView+Extensions.swift:10]: This is to fix a warning that was showing up

File: Sources/StreamChatUI/ChatThread/ChatThreadVC.swift [hunk 3]
```diff
@@ -276,7 +276,12 @@ open class ChatThreadVC: _ViewController,
         footerViewForMessage message: ChatMessage,
         at indexPath: IndexPath
     ) -> ChatMessageDecorationView? {
-        nil
+        guard components.threadRepliesCounterEnabled, message == messages.last else {
+            return nil
+        }
+        let repliesCounterDecorationView = components.threadRepliesCounterDecorationView.init()
+        repliesCounterDecorationView.content = L10n.Message.Thread.Replies.count(message.replyCount)
```
Comment (polqf) [inline Sources/StreamChatUI/ChatThread/ChatThreadVC.swift:283]: I know this is following the current pattern, but maybe this is something we should be discussing at some point. Not for this PR, though.

File: Tests/StreamChatUITests/SnapshotTests/ChatThread/ChatThreadVC_Tests.swift [hunk 4]
```diff
@@ -99,4 +99,77 @@ final class ChatThreadVC_Tests: XCTestCase {
         XCTAssertTrue(vc.messageListVC.listView is TestMessageListView)
         XCTAssertTrue(vc.messageComposerVC.composerView is TestComposerView)
     }
+
+    // MARK: - chatMessageListVC(_:footerViewForMessage:at)
```
Comment (polqf) [inline Tests/StreamChatUITests/SnapshotTests/ChatThread/ChatThreadVC_Tests.swift:103]: The snapshots showing "No replies" but actually having replies are a bit misleading. Shouldn't we actually have tests for whenever the thread is empty, where the footer should not be shown?

File: Sources/StreamChatUI/ChatThread/ChatThreadVC.swift (hunk 3 above)
Comment (ipavlidakis) [reply Sources/StreamChatUI/ChatThread/ChatThreadVC.swift:283]: What part are you referring to? The Localisation?

File: Tests/StreamChatUITests/SnapshotTests/ChatThread/ChatThreadVC_Tests.swift (hunk 4 above)
Comment (ipavlidakis) [reply Tests/StreamChatUITests/SnapshotTests/ChatThread/ChatThreadVC_Tests.swift:103]: The footer (at least on the React Native app) is visible even when there are no replies. I'll also discuss it with the designer team. 

We have tests that show the empty state `test_emptyAppearance`. 

I'll go ahead and update the tests that mistakenly show `NO REPLIES` when there are some

Comment (martinmitrevski) [reply Tests/StreamChatUITests/SnapshotTests/ChatThread/ChatThreadVC_Tests.swift:103]: Hm, tests like https://github.com/GetStream/stream-chat-swift/pull/2509/files#diff-7e8c52136701589324f567b355d9e0ebd7e7f073cf0b058813fa794e68dd2165 are a bit misleading, the "No replies" shouldn't be displayed there.

File: Sources/StreamChatUI/ChatThread/ChatThreadVC.swift (hunk 3 above)
Comment (polqf) [reply Sources/StreamChatUI/ChatThread/ChatThreadVC.swift:283]: Nope, the way we are passing data to the views. IMO this view should just get a model that contains the replies count. The responsibility of formatting the data should be in the (view) model of the view itself.

**Pull request merged**